src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeWalker.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using Type = Microsoft.VisualStudio.Debugger.Metadata.Type;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal struct TypeWalker
+    {
+        private readonly Type _type;
+
+        public TypeWalker(Type type)
+        {
+            _type = type;
+        }
+
+        public TypeEnumerator GetEnumerator()
+        {
+            return new TypeEnumerator(_type);
+        }
+
+        internal struct TypeEnumerator : IEnumerator, IDisposable
+        {
```

Comment:
IEnumerator<Type> (or drop the interface altogether) and change Current to return Type.